src/index.ts

```diff
@@ -109,10 +113,14 @@ module.exports = (context = {}) => {
             loader
         ]));
 
-        environment.on('template', (name: string, from: TwingSource) => {
-            this.addDependency(environment.getLoader().resolve(name, from));
+        environment.on('template', async (name: string, from: TwingSource) => {
+            this.addDependency(await environment.getLoader().resolve(name, from));
```

Comment:
This will create a race condition leading to the callback being called _before_ all `addDependency()` is called.

See https://github.com/hason-contributions/twing-loader/pull/1/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R154-R181 on how to await for all the `this.addDependency()` calls before calling the callback.

Reply:
Thanks. I'll add a test that trigger the issue and will fix it based on your example.